File: src/test/scala/photos/brooklyn/practice/codingexercises/scala/leetcode/ScoreOfParenthesesTest.scala
package photos.brooklyn.practice.codingexercises.scala.leetcode

import org.scalatest.{FlatSpec, Matchers}

import ScoreOfParentheses._

class ScoreOfParenthesesTest extends FlatSpec with Matchers {

  behavior of "ScoreOfParenthesesTest"

  it should "walkParentheses1" in {
    scoreOfParentheses("()") shouldEqual 1
  }

  it should "walkParenthesesEnclosed" in {
    scoreOfParentheses("(())") shouldEqual 2
  }

  it should "walkParenthesesDouble" in {
    scoreOfParentheses("()()") shouldEqual 2
  }

  it should "walkParenthesesComplex1" in {
    scoreOfParentheses("(()(()))") shouldEqual 6
  }

  it should "walkParenthesesComplex2" in {
    scoreOfParentheses("(()())()") shouldEqual 5
  }

  it should "walkParenthesesTwoEnclosed" in {
    scoreOfParentheses("(())(())") shouldEqual 4
  }

}

File: test/java/photos/brooklyn/practice/codingexercises/java/leetcode/solutions/ScoreOfParenthesesTest.java
package photos.brooklyn.practice.codingexercises.java.leetcode.solutions;

import org.junit.Test;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.*;

public class ScoreOfParenthesesTest {

    @Test
    public void scoreOfParenthesesSimplest() {
        assertThat(new ScoreOfParentheses().scoreOfParentheses("()"), is(1));
    }
    @Test
    public void scoreOfParenthesesEnclosed() {
        assertThat(new ScoreOfParentheses().scoreOfParentheses("(())"), is(2));
    }

    @Test
    public void scoreOfParenthesesDouble() {
        assertThat(new ScoreOfParentheses().scoreOfParentheses("()()"), is(2));
    }

    @Test
    public void scoreOfParenthesesComplex1() {
        assertThat(new ScoreOfParentheses().scoreOfParentheses("(()(()))"), is(6));
    }

    @Test
    public void scoreOfParenthesesComplex2() {
        assertThat(new ScoreOfParentheses().scoreOfParentheses("(()())()"), is(5));
    }
}